### PR TITLE
[Overflow-275] Update Question Detail API

### PR DESCRIPTION
**Pull request by nicoleamber**
## Backlog Link
https://framgiaph.backlog.com/view/SUN_OVERFLOW-275

## Commands

_Creating a private question_

`php artisan tinker`
- `Team::first()->members`
-- Select any user_id from the members
- `App\Models\Question::create(['title' => 'Team Question', 'content' => 'This is a private question', 'user_id' => 1, 'is_public' => false, 'team_id' => 1]);`
-- user_id should be what you selected from the team members

## Pre-conditions
- Create a private question
- Get the access token of a user who is a member of the first team and a user who is not a member

## Expected Output
Only users who are members of the team can access private questions. 

## Screenshots
_User 4 is a member of Team 1_
image
image

_User 31 is NOT a member of Team 1_
image
image

### CODE REVIEW

File: backend/app/GraphQL/Queries/Question.php [hunk 1]
```diff
@@ -31,10 +33,21 @@ public function __invoke($_, array $args)
 
         $question = $questionQuery->where('slug', $args['slug'])->first();
 
-        if ($args['shouldAddViewCount']) {
-            $question->update(['views_count' => $question->views_count + 1]);
+        $checker = true;
+
+        if (! $question->is_public) {
+            $team = Team::findOrFail($question->team_id);
```
Comment (jescabillas) [inline backend/app/GraphQL/Queries/Question.php]: Please use try and catch when using findOrFail so the fail will be captured with custom exceptions that return the correct error message

File: backend/app/GraphQL/Queries/Question.php [hunk 2]
```diff
@@ -31,10 +33,21 @@ public function __invoke($_, array $args)
...
         }
 
-        return $question;
+        if ($checker) {
+            if ($args['shouldAddViewCount']) {
+                $question->update(['views_count' => $question->views_count + 1]);
+            }
+
+            return $question;
+        }
```
Comment (kenorenzsun) [inline backend/app/GraphQL/Queries/Question.php]: Please add error message

**Review by kenorenzsun — APPROVED** [review]

**Review by jescabillas — COMMENTED** [review]
Review: 1
Comments: 1
Iteration: 1

**Review by jescabillas — APPROVED** [review]
Review: 2
Comments: 1
Iteration: 1

**Pull request merged**